gabbi/runner.py

```diff
@@ -89,26 +96,48 @@ def run():
              'form package.module or package.module:class.'
     )
 
-    args = parser.parse_args()
+    args = parser.parse_args(argv[1:])
     host, port, prefix, force_ssl = utils.host_info_from_target(
         args.target, args.prefix)
 
-    # Initialize response handlers.
-    handler_objects = initialize_handlers(args.response_handlers)
+    response_handlers = initialize_handlers(args.response_handlers)
 
-    data = utils.load_yaml(handle=sys.stdin)
+    failfast = args.failfast
+    failure = False
+    for input_file in input_files:
+        params = (response_handlers, host, port, prefix, force_ssl, failfast)
+        # XXX(FND): special-casing; use generic stream detection instead?
+        if input_file is sys.stdin:
+            success = execute(input_file, *params)
+        else:  # file path
+            with open(input_file, 'r') as fh:
+                success = execute(fh, *params)
+
+        if not failure:  # once failed, this is considered immutable
+            failure = not success
+        if failure and failfast:
+            break
+
+    sys.exit(failure)
+
+
+def execute(handle, response_handlers, host, port, prefix,  # TODO(FND): rename
```

Comment:
either accept the name or change it, do not leave for later. If you're trying to disambiguate, try `execute_tests` or `execute_test_file` or things in that vein